### PR TITLE
[SPARK-11743][SQL] Move the test for arrayOfUDT

### PR DESCRIPTION
**Pull request by viirya**
A following pr for #9712. Move the test for arrayOfUDT.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/RowEncoderSuite.scala [hunk 1]
```diff
@@ -128,19 +128,8 @@ class RowEncoderSuite extends SparkFunSuite {
...
-
-    val encoder = RowEncoder(schema)
-
-    val input: Row = Row(Seq(new ExamplePoint(0.1, 0.2), new ExamplePoint(0.3, 0.4)))
-    val row = encoder.toRow(input)
-    val convertedBack = encoder.fromRow(row)
-    assert(input.getSeq[ExamplePoint](0) == convertedBack.getSeq[ExamplePoint](0))
-  }
+      .add("structOfUDT", structOfUDT)
+      .add("arrayOfUDT", arrayOfUDT))
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/RowEncoderSuite.scala]: can we move it to the array section? i.e. the `encodeDecodeTest` that has a lot of `arrayOfxxx`

Comment (viirya) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/RowEncoderSuite.scala]: ok. it is more proper.

**Comment by cloud-fan** [comment]
LGTM, pending tests.

**Comment by rxin** [comment]
Thanks - I've merged this.

**Pull request closed without merging**